Source/Cache/BlobCache.swift

```diff
@@ -104,28 +102,66 @@ class BlobCache: DictionaryCache {
     
     private func  loadBlobFromCloud(for ref: BlobIdentifier)
     {
+        if let dataTask = self.dataTasks[ref] {
+            dataTask.resume()
+            return
+        }
+        
         let hexRef = ref.hexEncodedString()
         
         // first 2 chars are directory
         let dir = String(hexRef.prefix(2))
         // rest ist filename
         let restIdx = hexRef.index(hexRef.startIndex, offsetBy:2)
         let rest = String(hexRef[restIdx...])
-
         
         var gsUrl = URL(string: "https://storage.cloud.google.com/mainnet-blobs/")!
         gsUrl.appendPathComponent(dir)
         gsUrl.appendPathComponent(rest)
         
-        // We are running in a background thread, this is safe
-        guard let data = try? Data(contentsOf: gsUrl), let image = UIImage(data: data) else {
-            return
+        let dataTask = URLSession.shared.dataTask(with: gsUrl) { [weak self] (data, response, error) in
+             
+            Log.optional(error)
+            
+            guard error == nil,
+                let httpResponse = response as? HTTPURLResponse,
+                httpResponse.statusCode == 200,
+                let data = data else {
+                // Cannot use these because the servers are not reliable
+                //let mimeType = httpResponse.mimeType,
+                // mimeType.hasPrefix("image") else {
+                
+                self?.dataTasks.removeValue(forKey: ref)
+                return
+            }
+            
+            guard let image = UIImage(data: data) else {
```

Comment:
I think we can remove this once we are sure we are getting the blob from google cloud bucket. I left this because I was getting HTML files.